eslint.config.mjs

```diff
@@ -39,11 +42,28 @@ export default [
   {
     files: ['**/*.ts', '**/*.tsx'],
     ...typeScriptPlugin,
+    ...typeScriptTypeCheckedPlugin,
+    plugins: {
+      ...typeScriptPlugin.plugins,
+      ...typeScriptTypeCheckedPlugin.plugins,
+      deprecation: fixupPluginRules(deprecationPlugin)
+    },
     rules: {
       ...typeScriptPlugin.rules,
+      ...typeScriptTypeCheckedPlugin.rules,
       '@typescript-eslint/explicit-function-return-type': 'off',
       '@typescript-eslint/explicit-module-boundary-types': 'off',
       '@typescript-eslint/no-explicit-any': 'off',
+      '@typescript-eslint/no-floating-promises': 'off',
+      '@typescript-eslint/no-unnecessary-condition': 'off',
+      '@typescript-eslint/no-unsafe-argument': 'off',
+      '@typescript-eslint/no-unsafe-assignment': 'off',
+      '@typescript-eslint/no-unsafe-call': 'off',
+      '@typescript-eslint/no-unsafe-member-access': 'off',
+      '@typescript-eslint/no-unsafe-return': 'off',
+      '@typescript-eslint/strict-boolean-expressions': 'off',
+      '@typescript-eslint/unbound-method': 'off',
+      // 'deprecation/deprecation': 'warn', // TODO: enable for production
```

Comment:
Currently uncommenting this rule results in ~800 warnings – all of which we intend to address throughout the `next` development sequence.